src/Sarif/Writers/ResultLogJsonWriter.cs

```diff
@@ -243,16 +270,17 @@ public void Dispose()
             }
             else
             {
-                if ((_writeConditions & Conditions.ResultsInitialized) == Conditions.ResultsInitialized)
+                if ((_writeConditions & Conditions.ResultsInitialized) == Conditions.ResultsInitialized &&
+                    (_writeConditions & Conditions.ResultsClosed) != Conditions.ResultsClosed)
                 {
```

Comment:
CloseResults is resilient to multiple calls to close by design. Perhaps we should change that.